### PR TITLE
Pawn Capture Logic

### PR DESCRIPTION
**Pull request by chadbaum**
This implements pawn capture in a way that plugs into the move method.  Pawn's valid move list has been expanded to accept a diagonal move forward either left or right but only if it is occupied by an enemy piece.  Then it's passed forward to the move method just like any other piece for move and capture processing.

### CODE REVIEW

File: spec/models/pawn_spec.rb [hunk 1]
```diff
@@ -114,4 +114,47 @@
...
+      expect(moved_pawn.y_position).to eq 4
+      expect(moved_pawn.moved).to eq true
+      victim.reload
+      expect(victim.x_position).to eq 2
+      expect(victim.y_position).to eq 3
+      expect(victim.captured).to eq false
+    end
+
+    it 'should return true and update position on capturing enemy' do
+      victim = game.pieces.create(
```
Comment (aleksgorbenko) [inline spec/models/pawn_spec.rb]: Could these variable creation be moved to the top using `let(:white_rook) do`?

Comment (chadbaum) [inline spec/models/pawn_spec.rb]: Will refactor all the factorygirls (i feel sexist saying it that way but oh well) in sep PR

Comment (aleksgorbenko) [inline spec/models/pawn_spec.rb]: Ok makes sense.

**Comment by Se7enB2st** [comment]
👍

**Comment by ZachBerman** [comment]
:+1:

**Pull request merged**